.github/workflows/rust.yml

```diff
@@ -21,128 +57,88 @@ concurrency:
 
 jobs:
   check:
-    runs-on: ubuntu-latest
     name: check
+    runs-on:
+      group: 8-cores_32GB_Ubuntu Group
     timeout-minutes: 60
-    env:
-      RUSTFLAGS: -D warnings
-      SCCACHE_GHA_ENABLED: "true"
-      RUSTC_WRAPPER: "sccache"
     steps:
       - uses: actions/checkout@v3
+      - uses: rui314/setup-mold@v1
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
       - uses: Swatinem/rust-cache@v2
-      - uses: mozilla-actions/sccache-action@v0.0.3
+        with:
+          shared-key: cargo-check
+          workspaces: |
+            .
+            examples/demo-prover
       - name: Run lint
         run: |
           if ! make lint ; then
             echo "Linting or formatting errors detected, please run 'make lint-fix' to fix it";
             exit 1
           fi
-
-  # TODO: Temporary before migration to RiscZero 0.15: https://github.com/Sovereign-Labs/sovereign-sdk/issues/338
-  #   After that demo-prover should be integrated into workspace
-  check-demo-prover:
-    runs-on: ubuntu-latest
-    name: check demo prover
-    timeout-minutes: 90
-    env:
-      RUSTFLAGS: -D warnings
-      SCCACHE_GHA_ENABLED: "true"
-      RUSTC_WRAPPER: "sccache"
-    steps:
-      - uses: actions/checkout@v3
-      - uses: Swatinem/rust-cache@v2
-      - uses: mozilla-actions/sccache-action@v0.0.3
-      - name: Run cargo check
-        run: cd examples/demo-prover && cargo check
-      - name: Run cargo fmt check
-        run: |
-          cd examples/demo-prover;
-          if ! cargo fmt --check ; then
-            echo "Formatting errors detected, please run 'cargo fmt' to fix it";
-            exit 1
-          fi
-
   # Check that every combination of features is working properly.
   hack:
+    name: features
+    # `cargo-hack` uses the same profile as `cargo check` and doesn't require
+    # building dependencies, only chceking them, so we can share caches
+    # effectively.
+    needs: check
     runs-on:
       group: 8-cores_32GB_Ubuntu Group
-    name: features
     timeout-minutes: 60
-    env:
-      RUSTFLAGS: -D warnings
```

Comment:
Same for warnings, it will allow to have "unused imports" when feature gated incorrectly